sensors/cmd/start.go

```diff
@@ -70,20 +45,194 @@ func Start() {
 		logger.Fatal("required environment variable 'POD_NAME' not defined")
 	}
 
+	kubeConfig, _ := os.LookupEnv(common.EnvVarKubeConfig)
+	restConfig, err := common.GetClientConfig(kubeConfig)
+	if err != nil {
+		logger.Fatalw("failed to get kubeconfig", zap.Error(err))
+	}
 	dynamicClient := dynamic.NewForConfigOrDie(restConfig)
+	kubeClient := kubernetes.NewForConfigOrDie(restConfig)
+
+	ctx := logging.WithLogger(signals.SetupSignalHandler(), logger)
+
+	manager := NewSensorContextManager(logger, kubeClient, dynamicClient, busConfig, ebSubject, hostname)
+	defer manager.Close()
 
-	logger = logger.With("sensorName", sensor.Name)
-	for name, value := range sensor.Spec.LoggingFields {
-		logger.With(name, value)
+	if err := manager.Start(ctx); err != nil {
+		logger.Fatalw("failed to start sensor context manager", zap.Error(err))
+	}
+
+	for {
+		select {
+		case f := <-manager.C:
+			if err := f(); err != nil {
+				logger.Fatalw("failed to listen to events", zap.Error(err))
+			}
+		case <-ctx.Done():
+			return
+		}
+	}
+}
+
+type SensorContextManager struct {
+	C               chan func() error
+	logger          *zap.SugaredLogger
+	kubeClient      kubernetes.Interface
+	dynamicClient   dynamic.Interface
+	eventBusConfig  *eventbusv1alpha1.BusConfig
+	eventBusSubject string
+	hostname        string
+}
+
+func NewSensorContextManager(logger *zap.SugaredLogger, kubeClient kubernetes.Interface, dynamicClient dynamic.Interface, eventBusConfig *eventbusv1alpha1.BusConfig, eventBusSubject, hostname string) *SensorContextManager {
+	return &SensorContextManager{
+		C:               make(chan func() error),
+		logger:          logger,
+		kubeClient:      kubeClient,
+		dynamicClient:   dynamicClient,
+		eventBusConfig:  eventBusConfig,
+		eventBusSubject: eventBusSubject,
+		hostname:        hostname,
+	}
+}
+
+func (scm *SensorContextManager) Start(ctx context.Context) error {
+	// SENSOR_PATH env variable is used to indicate if this sensor definition
+	// has opted in to live reload
+	path, liveReload := os.LookupEnv("SENSOR_PATH")
```

Comment:
I changed my branch to use `common.SensorLiveReloadMountPath,` as the env var name

Reply:
Nice! Once we merge this I will update this line

Reply:
Added back log statements [here](https://github.com/intuit-data-os/argo-events/pull/29/files#diff-7c6e3f9dab2b101aae1efe37888503488a33f5705363b48d47a6533f56e1ec85R107-R112)